### PR TITLE
Remove unused ExecutionContexts in blaze-core

### PR DESCRIPTION
**Pull request by rossabaker**
Discovered during #6098.

### CODE REVIEW

**Comment by rossabaker** [comment]
I ran MiMa locally. [shakes fist]

File: build.sbt [hunk 1]
```diff
@@ -726,6 +726,12 @@ lazy val blazeServer = libraryProject("blaze-server")
         .exclude[DirectMissingMethodProblem]("org.http4s.blaze.server.BlazeServerBuilder.this"),
       ProblemFilters
         .exclude[DirectMissingMethodProblem]("org.http4s.blaze.server.WebSocketDecoder.this"),
+      ProblemFilters
+        .exclude[DirectMissingMethodProblem]("org.http4s.blazecore.util.BodylessWriter.this"),
+      ProblemFilters
+        .exclude[DirectMissingMethodProblem]("org.http4s.blazecore.util.BodylessWriter.ec"),
+      ProblemFilters
+        .exclude[DirectMissingMethodProblem]("org.http4s.blazecore.util.EntityBodyWriter.ec"),
```
Comment (armanbilge) [inline build.sbt:734]: I think these are Scala 3 only?

Comment (rossabaker) [reply build.sbt:734]: They definitely got changed on every version, but they're private.  Do I have it backward?  Is it Scala 3 that enforces privates and Scala 2 lets them through?  Maybe that's why I didn't see this locally.

**Comment by rossabaker** [comment]
There is more we could do here.  Some of the EC usage left here is a good candidate for `ExecutionContext.parasitic`.  We already have a `trampoline` and a `directec` in blaze that I'm reluctant to touch, but I'm also not sure they reflect modern usage.

File: build.sbt (hunk 1 above)
Comment (armanbilge) [reply build.sbt:734]: Mima doesn't know how to recognize package-privates on Scala 3 yet*, that's why we never see these locally.

* Actually It's fixed, but not released 😕 
- https://github.com/lightbend/mima/issues/682

**Pull request merged**